m4opt/constraints/_airmass.py

```diff
@@ -0,0 +1,51 @@
+import numpy as np
+from astropy import units as u
+
+from ._altitude import AltitudeConstraint
+
+
+class AirmassConstraint(AltitudeConstraint):
+    """
+    Constrains the airmass of a target by converting airmass limits to altitude limits.
+
+    The airmass is approximated as the secant of the zenith angle, which allows
+    for an equivalent formulation in terms of altitude constraints. This class
+    extends `AltitudeConstraint` and internally converts the given airmass
+    limits into corresponding altitude limits.
+
+    Parameters
+    ----------
+    max_airmass : float
+        Maximum airmass of the target (corresponds to minimum altitude).
+    min_airmass : float,
+        Minimum airmass of the target (corresponds to maximum altitude).
+        Default is `1` (the zenith).
+
+    Notes
+    -----
+    - The conversion from airmass to altitude follows:
+        `altitude = arcsin(1 / airmass)`, assuming a standard atmosphere.
+
+    Examples
+    --------
+    >>> from astropy.coordinates import EarthLocation, SkyCoord
+    >>> from astropy.time import Time
+    >>> from astropy import units as u
+    >>> from m4opt.constraints import AirmassConstraint
+    >>> time = Time("2017-08-17T00:41:04Z")
+    >>> target = SkyCoord.from_name("NGC 4993")
+    >>> location = EarthLocation.of_site("Rubin Observatory")
+    >>> constraint = AirmassConstraint(max_airmass=3, min_airmass=1)
+    >>> constraint(location, target, time)
+    np.True_
+    """
+
+    def __init__(
+        self,
+        max_airmass: float,
+        min_airmass: float = 1.0,
```

Comment:
Why do we need to support a minimum airmass? I don't see the use case.

Reply:
In the past, we have wanted to take calibration standards at a variety of air masses, so one could use this to ensure some high airmass observations.

Reply:
I see. Does it still make sense that minimum airmass is optional?

Reply:
Sorry, what I mean to ask is: does it still make sense that maximum airmass is required?

Reply:
I think a sensible default is ok.

Reply:
Can we add:

```python
min_airmass, max_airmass = sorted([min_airmass, max_airmass])
```

so that it doesn't matter the order in which these arguments are passed? That would allow us to support both of the following:

```python
AirmassConstraint(2.5)
AirmassConstraint(3, 4)
```